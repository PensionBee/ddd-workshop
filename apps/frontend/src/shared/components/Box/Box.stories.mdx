import { Meta, Story, Canvas, ArgsTable } from "@storybook/addon-docs";
import { Box } from "./Box";

<Meta
  title="Shared/Box/Docs"
  parameters={{
    viewMode: "docs",
    previewTabs: {
      canvas: { hidden: true },
    },
  }}
/>

# Box

## Props

<ArgsTable of={Box} />
