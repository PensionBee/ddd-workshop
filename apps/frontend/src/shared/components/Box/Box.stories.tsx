import type { ComponentStory, ComponentMeta } from "@storybook/react";
import type { BoxProps } from "./Box";
import { Box } from "./Box";

export default {
  title: "Shared/Box",
  component: Box,
  parameters: {
    viewMode: "story",
    previewTabs: {
      "storybook/docs/panel": { hidden: true },
    },
  },
} as ComponentMeta<typeof Box>;

const BoxTemplate: ComponentStory<typeof Box> = (args: BoxProps) => {
  return (
    <Box {...args}>This is a Box - You can style me however you need to</Box>
  );
};

export const Default = BoxTemplate.bind({});
