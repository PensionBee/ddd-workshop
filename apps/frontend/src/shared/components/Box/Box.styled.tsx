import type React from "react";
import styled, { css } from "styled-components";
import * as styledHelpers from "@shared/styling/styledComponentHelpers";
import type { ThemeColors, ThemeGradients } from "@shared/styling/theme";
import { theme } from "@shared/styling/theme";
import type { BreakpointValues } from "@shared/utils/types";
import type {
  MediaAlignType,
  MediaBorderRadiusType,
  MediaBorderStyleType,
  MediaBorderType,
  MediaBoxShadowType,
  MediaCursorType,
  MediaDirectionType,
  MediaDisplayType,
  MediaFlexItemOrderType,
  MediaFlexType,
  MediaGapType,
  MediaHeightType,
  MediaJustifyType,
  MediaMarginType,
  MediaNoWrapType,
  MediaOpacityType,
  MediaOverflowType,
  MediaPaddingType,
  MediaPositionType,
  MediaPositionValue,
  MediaWidthType,
  MediaZIndex,
} from "@shared/styling/styledComponentHelpers";

export type StyledBoxProps = {
  as: React.ElementType;
  // Layout props
  $d: BreakpointValues<MediaDisplayType>;
  $h: BreakpointValues<MediaHeightType>;
  $hMin: BreakpointValues<MediaHeightType>;
  $hMax: BreakpointValues<MediaHeightType>;
  $overflow: BreakpointValues<MediaOverflowType>;
  $overflowX: BreakpointValues<MediaOverflowType>;
  $overflowY: BreakpointValues<MediaOverflowType>;
  $w: BreakpointValues<MediaWidthType>;
  $wMax: BreakpointValues<MediaWidthType>;
  $wMin: BreakpointValues<MediaWidthType>;
  // Flex container props
  $align: BreakpointValues<MediaAlignType>;
  $colGap: BreakpointValues<MediaGapType>;
  $rowGap: BreakpointValues<MediaGapType>;
  $direction: BreakpointValues<MediaDirectionType>;
  $flex: BreakpointValues<MediaFlexType>;
  $justify: BreakpointValues<MediaJustifyType>;
  $nowrap: BreakpointValues<MediaNoWrapType>;
  // Flex item props
  $flexItemOrder: BreakpointValues<MediaFlexItemOrderType>;
  // Space props
  $m: BreakpointValues<MediaMarginType>;
  $mx: BreakpointValues<MediaMarginType>;
  $my: BreakpointValues<MediaMarginType>;
  $mb: BreakpointValues<MediaMarginType>;
  $ml: BreakpointValues<MediaMarginType>;
  $mr: BreakpointValues<MediaMarginType>;
  $mt: BreakpointValues<MediaMarginType>;
  $p: BreakpointValues<MediaPaddingType>;
  $px: BreakpointValues<MediaPaddingType>;
  $py: BreakpointValues<MediaPaddingType>;
  $pb: BreakpointValues<MediaPaddingType>;
  $pl: BreakpointValues<MediaPaddingType>;
  $pr: BreakpointValues<MediaPaddingType>;
  $pt: BreakpointValues<MediaPaddingType>;
  // Position props
  $position: BreakpointValues<MediaPositionType>;
  $bottom: BreakpointValues<MediaPositionValue>;
  $left: BreakpointValues<MediaPositionValue>;
  $right: BreakpointValues<MediaPositionValue>;
  $top: BreakpointValues<MediaPositionValue>;
  $z: BreakpointValues<MediaZIndex>;
  // Border props
  $backgroundColor: ThemeColors;
  $backgroundGradient?: ThemeGradients;
  $borderColor: ThemeColors;
  $border: BreakpointValues<MediaBorderType>;
  $borderBottom: BreakpointValues<MediaBorderType>;
  $borderLeft: BreakpointValues<MediaBorderType>;
  $borderRight: BreakpointValues<MediaBorderType>;
  $borderTop: BreakpointValues<MediaBorderType>;
  $borderStyle: BreakpointValues<MediaBorderStyleType>;
  $borderRadius: BreakpointValues<MediaBorderRadiusType>;
  // Effect props
  $boxShadow: BreakpointValues<MediaBoxShadowType>;
  $opacity: BreakpointValues<MediaOpacityType>;
  // Animation props
  $transition?: string;
  // Action props
  $cursor: BreakpointValues<MediaCursorType>;
  $isClickable: boolean;
};

export const StyledBox = styled.div<StyledBoxProps>`
  position: relative;
  box-sizing: border-box;
  ${({ $backgroundColor }) =>
    css`
      background-color: ${theme.colors[$backgroundColor]};
    `}
  ${({ $backgroundGradient }) =>
    $backgroundGradient &&
    css`
      background-image: linear-gradient(
        ${theme.gradients[$backgroundGradient].join(", ")}
      );
    `}
  border-color: ${({ $borderColor }) => theme.colors[$borderColor]};
  transition: ${({ $transition }) => $transition};
  ${({ $isClickable }) =>
    $isClickable &&
    css`
      cursor: pointer;
      outline: 3px solid transparent;
      &:hover {
        outline: 3px solid ${() => theme.colors.grey3};
      }
      &:focus-visible {
        outline: 3px solid ${() => theme.colors.blue100};
      }
    `}
  ${(props) =>
    styledHelpers.createMediaStyles(
      (breakpoint) => css`
        // Layout props
        ${styledHelpers.mediaDisplay(breakpoint, props.$d)}
        ${styledHelpers.mediaHeight(breakpoint, props.$h)}
        ${styledHelpers.mediaMaxHeight(breakpoint, props.$hMax)}
        ${styledHelpers.mediaMaxWidth(breakpoint, props.$wMax)}
        ${styledHelpers.mediaMinWidth(breakpoint, props.$wMin)}
        ${styledHelpers.mediaMinHeight(breakpoint, props.$hMin)}
        ${styledHelpers.mediaOverflow(breakpoint, props.$overflow)}
        ${styledHelpers.mediaOverflowX(breakpoint, props.$overflowX)}
        ${styledHelpers.mediaOverflowY(breakpoint, props.$overflowY)}
        ${styledHelpers.mediaWidth(breakpoint, props.$w)}
        ${styledHelpers.mediaMaxWidth(breakpoint, props.$wMax)}
        ${styledHelpers.mediaMinWidth(breakpoint, props.$wMin)}
        // Flex container props
        ${styledHelpers.mediaAlign(breakpoint, props.$align)}
        ${styledHelpers.mediaColGap(breakpoint, props.$colGap)}
        ${styledHelpers.mediaDirection(breakpoint, props.$direction)}
        ${styledHelpers.mediaFlex(breakpoint, props.$flex)}
        ${styledHelpers.mediaJustify(breakpoint, props.$justify)}
        ${styledHelpers.mediaNoWrap(breakpoint, props.$nowrap)}
        ${styledHelpers.mediaRowGap(breakpoint, props.$rowGap)}
        // Flex item props
        ${styledHelpers.mediaFlexItemOrder(breakpoint, props.$flexItemOrder)}
        // Space props
        ${styledHelpers.mediaMargin(breakpoint, props.$m)}
        ${styledHelpers.mediaMarginX(breakpoint, props.$mx)}
        ${styledHelpers.mediaMarginY(breakpoint, props.$my)}
        ${styledHelpers.mediaMarginBottom(breakpoint, props.$mb)}
        ${styledHelpers.mediaMarginLeft(breakpoint, props.$ml)}
        ${styledHelpers.mediaMarginRight(breakpoint, props.$mr)}
        ${styledHelpers.mediaMarginTop(breakpoint, props.$mt)}
        ${styledHelpers.mediaPadding(breakpoint, props.$p)}
        ${styledHelpers.mediaPaddingX(breakpoint, props.$px)}
        ${styledHelpers.mediaPaddingY(breakpoint, props.$py)}
        ${styledHelpers.mediaPaddingBottom(breakpoint, props.$pb)}
        ${styledHelpers.mediaPaddingLeft(breakpoint, props.$pl)}
        ${styledHelpers.mediaPaddingRight(breakpoint, props.$pr)}
        ${styledHelpers.mediaPaddingTop(breakpoint, props.$pt)}
        // Position props
        ${styledHelpers.mediaBottom(breakpoint, props.$bottom)}
        ${styledHelpers.mediaLeft(breakpoint, props.$left)}
        ${styledHelpers.mediaPosition(breakpoint, props.$position)}
        ${styledHelpers.mediaRight(breakpoint, props.$right)}
        ${styledHelpers.mediaTop(breakpoint, props.$top)}
        ${styledHelpers.mediaZIndex(breakpoint, props.$z)}
        // Style / effect props
        ${styledHelpers.mediaBorder(breakpoint, props.$border)}
        ${styledHelpers.mediaBorderBottom(breakpoint, props.$borderBottom)}
        ${styledHelpers.mediaBorderLeft(breakpoint, props.$borderLeft)}
        ${styledHelpers.mediaBorderRight(breakpoint, props.$borderRight)}
        ${styledHelpers.mediaBorderTop(breakpoint, props.$borderTop)}
        ${styledHelpers.mediaBorderStyle(breakpoint, props.$borderStyle)}
        ${styledHelpers.mediaBorderRadius(breakpoint, props.$borderRadius)}
        ${styledHelpers.mediaBoxShadow(breakpoint, props.$boxShadow)}
        ${styledHelpers.mediaOpacity(breakpoint, props.$opacity)}
        // Action props
        ${styledHelpers.mediaCursor(breakpoint, props.$cursor)}
      `
    )}
`;
