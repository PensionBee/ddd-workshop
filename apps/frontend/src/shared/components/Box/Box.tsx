import type { ValueOrBreakpointValues } from "@shared/utils/types";
import { asBreakpointObject } from "@shared/utils/asBreakpointObject";
import type { ThemeColors, ThemeGradients } from "@shared/styling/theme";
import { StyledBox } from "./Box.styled";
import React from "react";
import type {
  MediaAlignType,
  MediaBorderRadiusType,
  MediaBorderStyleType,
  MediaBorderType,
  MediaBoxShadowType,
  MediaCursorType,
  MediaDirectionType,
  MediaDisplayType,
  MediaFlexItemOrderType,
  MediaFlexType,
  MediaGapType,
  MediaHeightType,
  MediaJustifyType,
  MediaMarginType,
  MediaNoWrapType,
  MediaOpacityType,
  MediaOverflowType,
  MediaPaddingType,
  MediaPositionType,
  MediaPositionValue,
  MediaWidthType,
  MediaZIndex,
} from "@shared/styling/styledComponentHelpers";

type BoxLayoutProps = {
  /** Display option */
  d?: ValueOrBreakpointValues<MediaDisplayType>;
  /** Height */
  h?: ValueOrBreakpointValues<MediaHeightType>;
  /** Maximum height */
  hMax?: ValueOrBreakpointValues<MediaHeightType>;
  /** Minimum height */
  hMin?: ValueOrBreakpointValues<MediaHeightType>;
  /** Overflow */
  overflow?: ValueOrBreakpointValues<MediaOverflowType>;
  /** Horizontal overflow */
  overflowX?: ValueOrBreakpointValues<MediaOverflowType>;
  /** Vertical overflow */
  overflowY?: ValueOrBreakpointValues<MediaOverflowType>;
  /** Width */
  w?: ValueOrBreakpointValues<MediaWidthType>;
  /** Maximum width */
  wMax?: ValueOrBreakpointValues<MediaWidthType>;
  /** Minimum width */
  wMin?: ValueOrBreakpointValues<MediaWidthType>;
};

type BoxFlexProps = {
  /** Flexbox secondary dimension item alignment */
  align?: ValueOrBreakpointValues<MediaAlignType>;
  /** Vertical space between elements */
  colGap?: ValueOrBreakpointValues<MediaGapType>;
  /** Flex direction */
  direction?: ValueOrBreakpointValues<MediaDirectionType>;
  /** Flex options */
  flex?: ValueOrBreakpointValues<MediaFlexType>;
  /** Flexbox primary dimension item alignment */
  justify?: ValueOrBreakpointValues<MediaJustifyType>;
  /** If wrap should be disabled */
  nowrap?: ValueOrBreakpointValues<MediaNoWrapType>;
  /** Horizontal space between elements */
  rowGap?: ValueOrBreakpointValues<MediaGapType>;
  /** Flex item order, if applicable [default = 0],  */
  flexItemOrder?: ValueOrBreakpointValues<MediaFlexItemOrderType>;
};

type BoxSpaceProps = {
  /** Margin in px */
  m?: ValueOrBreakpointValues<MediaMarginType>;
  /** Horizontal margin in px */
  mx?: ValueOrBreakpointValues<MediaMarginType>;
  /** Vertical margin in px */
  my?: ValueOrBreakpointValues<MediaMarginType>;
  /** Margin bottom in px */
  mb?: ValueOrBreakpointValues<MediaMarginType>;
  /** Margin left in px */
  ml?: ValueOrBreakpointValues<MediaMarginType>;
  /** Margin right in px */
  mr?: ValueOrBreakpointValues<MediaMarginType>;
  /** Margin top in px */
  mt?: ValueOrBreakpointValues<MediaMarginType>;
  /** Padding in px */
  p?: ValueOrBreakpointValues<MediaPaddingType>;
  /** Padding horizontal in px */
  px?: ValueOrBreakpointValues<MediaPaddingType>;
  /** Padding vertical in px */
  py?: ValueOrBreakpointValues<MediaPaddingType>;
  /** Padding bottom in px */
  pb?: ValueOrBreakpointValues<MediaPaddingType>;
  /** Padding left in px */
  pl?: ValueOrBreakpointValues<MediaPaddingType>;
  /** Padding right in px */
  pr?: ValueOrBreakpointValues<MediaPaddingType>;
  /** Padding top in px */
  pt?: ValueOrBreakpointValues<MediaPaddingType>;
};

type BoxPositionProps = {
  /** Bottom position offset */
  bottom?: ValueOrBreakpointValues<MediaPositionValue>;
  /** Left position offset */
  left?: ValueOrBreakpointValues<MediaPositionValue>;
  /** Position */
  position?: ValueOrBreakpointValues<MediaPositionType>;
  /** Right position offset */
  right?: ValueOrBreakpointValues<MediaPositionValue>;
  /** Top position offset */
  top?: ValueOrBreakpointValues<MediaPositionValue>;
  /** Z Index */
  z?: ValueOrBreakpointValues<MediaZIndex>;
};

type BoxStyleEffectProps = {
  /** Background color */
  backgroundColor?: ThemeColors;
  /** Background gradient */
  backgroundGradient?: ThemeGradients;
  /** Border color */
  borderColor?: ThemeColors;
  /** Border in px */
  border?: ValueOrBreakpointValues<MediaBorderType>;
  /** Border bottom in px */
  borderBottom?: ValueOrBreakpointValues<MediaBorderType>;
  /** Border left in px */
  borderLeft?: ValueOrBreakpointValues<MediaBorderType>;
  /** Border right in px */
  borderRight?: ValueOrBreakpointValues<MediaBorderType>;
  /** Border top in px */
  borderTop?: ValueOrBreakpointValues<MediaBorderType>;
  /** Border radius in px */
  borderRadius?: ValueOrBreakpointValues<MediaBorderRadiusType>;
  /** Border style */
  borderStyle?: ValueOrBreakpointValues<MediaBorderStyleType>; //keyof typeof theme.borderStyle;
  /** box shadow definition */
  boxShadow?: ValueOrBreakpointValues<MediaBoxShadowType>;
  /** Opacity */
  opacity?: ValueOrBreakpointValues<MediaOpacityType>;
};

type BoxAnimationProps = {
  /** Transition definition */
  transition?: string;
};

type BoxActionProps = {
  /** Cursor type */
  cursor?: MediaCursorType;
  /** for property used by labels */
  htmlFor?: string;
};

export type BoxProps = BoxLayoutProps &
  BoxFlexProps &
  BoxSpaceProps &
  BoxPositionProps &
  BoxStyleEffectProps &
  BoxAnimationProps &
  BoxActionProps & {
    /** React children */
    children?: React.ReactNode;
    /** Element to render as */
    as?: React.ElementType;
  };

export const Box = React.forwardRef<HTMLElement, BoxProps>(
  (
    {
      children,
      as = "div",
      // Layout props
      d,
      h,
      hMax,
      hMin,
      overflow,
      overflowX,
      overflowY,
      w,
      wMax,
      wMin,
      // Flex container props
      align,
      colGap,
      direction,
      flex,
      justify,
      nowrap = false,
      rowGap,
      // Flex item props
      flexItemOrder,
      // Space props
      m,
      mb,
      ml,
      mr,
      mt,
      mx,
      my,
      p,
      pb,
      pl,
      pr,
      pt,
      px,
      py,
      // Position props
      bottom,
      left,
      position,
      right,
      top,
      z,
      // Style / effect props
      backgroundColor,
      backgroundGradient,
      border = 0,
      borderStyle = "solid",
      borderRadius,
      borderBottom,
      borderColor,
      borderLeft,
      borderRight,
      borderTop,
      boxShadow,
      opacity,
      // Animation props
      transition,
      // Action props
      cursor,
      ...props
    },
    ref
  ) => {
    const styledProps = {
      as: as,
      // Layout props
      $d: asBreakpointObject(d),
      $h: asBreakpointObject(h),
      $hMax: asBreakpointObject(hMax),
      $hMin: asBreakpointObject(hMin),
      $overflow: asBreakpointObject(overflow),
      $overflowX: asBreakpointObject(overflowX),
      $overflowY: asBreakpointObject(overflowY),
      $w: asBreakpointObject(w),
      $wMax: asBreakpointObject(wMax),
      $wMin: asBreakpointObject(wMin),
      // Flex container props
      $align: asBreakpointObject(align),
      $colGap: asBreakpointObject(colGap),
      $direction: asBreakpointObject(direction),
      $flex: asBreakpointObject(flex),
      $justify: asBreakpointObject(justify),
      $nowrap: asBreakpointObject(nowrap),
      $rowGap: asBreakpointObject(rowGap),
      // Flex item props
      $flexItemOrder: asBreakpointObject(flexItemOrder),
      // Space props
      $m: asBreakpointObject(m),
      $mx: asBreakpointObject(mx),
      $my: asBreakpointObject(my),
      $mb: asBreakpointObject(mb),
      $ml: asBreakpointObject(ml),
      $mr: asBreakpointObject(mr),
      $mt: asBreakpointObject(mt),
      $p: asBreakpointObject(p),
      $px: asBreakpointObject(px),
      $py: asBreakpointObject(py),
      $pb: asBreakpointObject(pb),
      $pl: asBreakpointObject(pl),
      $pr: asBreakpointObject(pr),
      $pt: asBreakpointObject(pt),
      // Position props
      $bottom: asBreakpointObject(bottom),
      $left: asBreakpointObject(left),
      $position: asBreakpointObject(position),
      $right: asBreakpointObject(right),
      $top: asBreakpointObject(top),
      $z: asBreakpointObject(z),
      // Style / effect props
      $backgroundColor: backgroundColor,
      $backgroundGradient: backgroundGradient,
      $borderColor: borderColor,
      $borderStyle: asBreakpointObject(borderStyle),
      $borderRadius: asBreakpointObject(borderRadius),
      $border: asBreakpointObject(border),
      $borderBottom: asBreakpointObject(borderBottom),
      $borderLeft: asBreakpointObject(borderLeft),
      $borderRight: asBreakpointObject(borderRight),
      $borderTop: asBreakpointObject(borderTop),
      $boxShadow: asBreakpointObject(boxShadow),
      $opacity: asBreakpointObject(opacity),
      // Animation props
      $transition: transition,
      // Action props
      $cursor: asBreakpointObject(cursor),
      $isClickable: as === "button" || as === "a",
    };
    return (
      <StyledBox ref={ref} {...styledProps} {...props}>
        {children}
      </StyledBox>
    );
  }
);

Box.displayName = "Box";
