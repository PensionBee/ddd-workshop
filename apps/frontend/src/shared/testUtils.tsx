import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import type { Breakpoints } from "./styling/breakpoints";
import { BREAKPOINTS } from "./styling/breakpoints";
import { theme } from "./styling/theme";
import { vi } from "vitest"

type ReactTestingLibraryProviderProps = {
  children: React.ReactNode;
};

// eslint-disable-next-line react-refresh/only-export-components
const ReactTestingLibraryProviders: React.FC<
  ReactTestingLibraryProviderProps
> = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const customRender = (ui: React.ReactElement, options?: any) => {
  return render(ui, { wrapper: ReactTestingLibraryProviders, ...options });
};

// re-export everything
// eslint-disable-next-line react-refresh/only-export-components
export * from "@testing-library/react";

// override render method
export { customRender as render };

export const sleep = (delay: number) => {
  return new Promise((resolve) => setTimeout(resolve, delay));
};

export const mockMediaMatches = (breakpoint: Breakpoints) => {
  Object.defineProperty(window, "matchMedia", {
    writable: true,
    value: (query: string) => {
      const breakpointValue = BREAKPOINTS[breakpoint];
      const queryValue = parseInt(query.replace(/^\D+/g, ""));
      const breakpointValues = [...Object.values(BREAKPOINTS)];
      if (!breakpointValues.includes(queryValue)) {
        throw new Error(
          `Invalid breakpoint value: ${queryValue}, this likely means there has been a change to the way that breakpoints are setup.`
        );
      }
      return {
        matches: queryValue <= breakpointValue,
        addEventListener: vi.fn(),
        removeEventListener: vi.fn(),
      };
    },
  });
};

export const routerRender = (ui: React.ReactNode, pathname = "/") => {
  return customRender(
    <MemoryRouter initialEntries={[{ pathname }]}>{ui}</MemoryRouter>
  );
};
