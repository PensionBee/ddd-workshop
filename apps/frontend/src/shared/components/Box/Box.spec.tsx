import { render } from "@shared/testUtils";
import { vi } from "vitest";
import { Box } from "./Box";

describe("Box", () => {
  const boxContent = "Box Content";
  describe("Default", () => {
    it("should render correctly", () => {
      const { getByText, asFragment } = render(<Box>{boxContent}</Box>);

      expect(getByText(boxContent)).toBeVisible();
      expect(asFragment()).toMatchSnapshot();
    });

    it("should render as elements", () => {
      const { container, asFragment } = render(
        <Box as="span">{boxContent}</Box>
      );

      expect(container.getElementsByTagName("span").length).toBe(1);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
