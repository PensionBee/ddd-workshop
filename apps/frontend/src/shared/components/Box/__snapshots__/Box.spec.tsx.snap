// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box Default should render as elements 1`] = `
<DocumentFragment>
  <span
    class="sc-bcXHqe bLlgzy"
  >
    Box Content
  </span>
</DocumentFragment>
`;

exports[`Box Default should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe bLlgzy"
  >
    Box Content
  </div>
</DocumentFragment>
`;
